aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -181,8 +181,18 @@ def export_tree(
             )
 
     # Add all the nodes contained within the specified groups
-    for group in given_groups:
-        for entry in group.nodes:
+    if given_group_entry_ids:
+
+        if not silent:
+            print('RETRIEVING NODES FROM GROUPS...')
+
+        # Use single query instead of given_group.nodes iterator for performance.
+        group_qb = orm.QueryBuilder().append(orm.Group, filters={'id': {'in': given_group_entry_ids}}, tag='groups')
+        # Could project on ['id', 'uuid', 'node_type'] for further performance enhancement
+        group_qb.append(orm.Node, with_group='groups', project=['*'])
+
+        for row in group_qb.all():
```

Comment:
Could you use group_qb.iterall() here? It might lead to further improvement because the results are fetched in batches and not all together (if the subsequent code also takes time this could speed up the time-to-solution).